### PR TITLE
Command argument mappings (and cooking categories)

### PR DESCRIPTION
**Pull request by Cypher121**
All the things needed for 1.19.3 QKL, mostly brigadier stuff

Not too sure about the getAsync name for messages, open to alternatives

### CODE REVIEW

File: mappings/net/minecraft/command/argument/RegistryEntryArgumentType.mapping [hunk 1]
```diff
@@ -0,0 +1,28 @@
+CLASS net/minecraft/unmapped/C_mymnmzsx net/minecraft/command/argument/RegistryEntryArgumentType
+	FIELD f_fkeeakna EXAMPLES Ljava/util/Collection;
+	FIELD f_mjmvcceu REGISTRY_ENTRY_NOT_FOUND_EXCEPTION Lcom/mojang/brigadier/exceptions/Dynamic2CommandExceptionType;
+	FIELD f_nhwrxdmt ENTITY_NOT_SUMMONABLE_EXCEPTION Lcom/mojang/brigadier/exceptions/DynamicCommandExceptionType;
+	FIELD f_qycsdcbt REGISTRY_ENTRY_INVALID_TYPE_EXCEPTION Lcom/mojang/brigadier/exceptions/Dynamic3CommandExceptionType;
```
Comment (eristhea) [inline mappings/net/minecraft/command/argument/RegistryEntryArgumentType.mapping]:
```suggestion
	FIELD f_qycsdcbt INVALID_REGISTRY_ENTRY_TYPE_EXCEPTION Lcom/mojang/brigadier/exceptions/Dynamic3CommandExceptionType;
```
small improvement to readability here

**Review by eristhea — APPROVED** [review]

File: mappings/net/minecraft/command/argument/MessageArgumentType.mapping [hunk 2]
```diff
@@ -1,6 +1,7 @@
 CLASS net/minecraft/unmapped/C_fbgfoxlq net/minecraft/command/argument/MessageArgumentType
 	FIELD f_gososwfv EXAMPLES Ljava/util/Collection;
 	METHOD getExamples getExamples ()Ljava/util/Collection;
+	METHOD m_hqcmqrcd getSignedMessageAsync (Lcom/mojang/brigadier/context/CommandContext;Ljava/lang/String;Ljava/util/function/Consumer;)V
```
Comment (IotaBread) [inline mappings/net/minecraft/command/argument/MessageArgumentType.mapping]: What does this do?

**Review by IotaBread — APPROVED** [review]

File: mappings/net/minecraft/command/argument/MessageArgumentType.mapping (hunk 2 above)
Comment (Cypher121) [reply mappings/net/minecraft/command/argument/MessageArgumentType.mapping]: Passes the SignedChatMessage to the Consumer in the last argument once a CompletableFuture made by one of the private methods resolves.

Comment (IotaBread) [reply mappings/net/minecraft/command/argument/MessageArgumentType.mapping]: How is this name based on mojmap and yarn? Plus parameter names
```suggestion
	METHOD m_hqcmqrcd resolveSignedMessage (Lcom/mojang/brigadier/context/CommandContext;Ljava/lang/String;Ljava/util/function/Consumer;)V
		ARG 0 context
		ARG 1 name
		ARG 2 callback
```

Comment (Cypher121) [reply mappings/net/minecraft/command/argument/MessageArgumentType.mapping]: "resolve" feels a bit ambiguous imho. I'll merge the suggestion for arg names.

**Pull request merged**